src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineReceiver.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\Serializer;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class DoctrineReceiver implements ReceiverInterface
+{
+    /** @var \Symfony\Component\Messenger\Transport\Doctrine\Connection */
+    private $connection;
+    /** @var \Symfony\Component\Messenger\Transport\Serialization\Serializer */
+    private $serializer;
+    /** @var bool */
+    private $shouldStop = false;
+
+    public function __construct(Connection $connection, Serializer $serializer = null)
+    {
+        $this->connection = $connection;
+        $this->serializer = $serializer ?? Serializer::create();
+    }
+
+    public function receive(callable $handler): void
+    {
+        while (!$this->shouldStop) {
+            $doctrineEnvelop = $this->connection->get();
+
+            if (null === $doctrineEnvelop) {
+                $handler(null);
+
+                usleep(200000);
+                if (\function_exists('pcntl_signal_dispatch')) {
```

Comment:
why not emerge method :) ?

Reply:
You mean to had a method with those 3 lines to avoid duplication at line 63 ?

```php

    private function emerge(): void
    {
        if (\function_exists('pcntl_signal_dispatch')) {
            pcntl_signal_dispatch();
        }
    }
```

For this class I just barely copied from https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php#L47 with some adaptations

Reply:
yes, but she would call herself more pcntlDispatch